client/trino-client/src/main/java/io/trino/client/FixJsonDataUtils.java

```diff
@@ -68,16 +68,13 @@ public static Iterable<List<Object>> fixData(List<Column> columns, List<List<Obj
             if (row.size() != typeHandlers.length) {
                 throw new IllegalArgumentException("row/column size mismatch");
             }
-            ArrayList<Object> newRow = new ArrayList<>(typeHandlers.length);
-            int column = 0;
-            for (Object value : row) {
+            for (int i = 0; i < row.size(); i++) {
+                Object value = row.get(i);
                 if (value != null) {
-                    value = typeHandlers[column].fixValue(value);
+                    row.set(i, typeHandlers[i].fixValue(value)); // update value in-place
```

Comment:
how can you assume here that provided list is modifiable. What is the purpose of this change?

Reply:
it is modifiable since Jackson returns it as ArrayList.

Reply:
The fact that you can does not mean you should. It is fragile. What are we gaining.